staging/src/k8s.io/client-go/tools/cache/listers.go

```diff
@@ -31,28 +31,63 @@ import (
 type AppendFunc func(interface{})
 
 func ListAll(store Store, selector labels.Selector, appendFn AppendFunc) error {
+	listOptions := metav1.ListOptions{}
+	listOptions.LabelSelector = selector.String()
+	listOptions.IncludeUninitialized = false
+	return ListAllWithOptions(store, listOptions, appendFn)
+}
+
+// ListAllWithOptions only respects the Selector and the IncludeUninitialized of the options.
+// AppendFunc is used to add a matching item to whatever list the caller is using
+func ListAllWithOptions(store Store, options metav1.ListOptions, appendFn AppendFunc) error {
+	includeUninitialized := options.IncludeUninitialized
+	selector, err := labels.Parse(options.LabelSelector)
+	if err != nil {
+		return err
+	}
 	for _, m := range store.List() {
 		metadata, err := meta.Accessor(m)
 		if err != nil {
 			return err
 		}
-		if selector.Matches(labels.Set(metadata.GetLabels())) {
-			appendFn(m)
+		if !selector.Matches(labels.Set(metadata.GetLabels())) {
+			continue
+		}
+		if !includeUninitialized && metadata.GetInitializers() != nil {
```

Comment:
metadata.GetInitializers() == nil means it is initialized? I have a hard time understanding this line. My assumption is if `includeUninitialized` is `false` then the function should behave exactly as before, right? Starting to combine conditions like this:

```
if !selector.Matches(labels.Set(metadata.GetLabels()))  || (!includeUninitialized && metadata.GetInitializers() != nil) {
 			continue
}
```
negating the condition would be

```
if selector.Matches(labels.Set(metadata.GetLabels()))  && (includeUninitialized || metadata.GetInitializers() == nil) {
 			appendFn(m)
}
```

now if we assume `includeUninitialized == false` we would have:


```
if selector.Matches(labels.Set(metadata.GetLabels()))  && metadata.GetInitializers() == nil {
 			appendFn(m)
}
```

This is different than original statement.

Reply:
>  if includeUninitialized is false then the function should behave exactly as before, right? 

The goal is to keep the returned results to be the same, but the internals might be different. In other words, this function should filter out uninitialized objects if `includeUninitialized == false`. So the final form of the condition is the desired behavior.